docs/dev/adding_instruments.rst

```diff
@@ -644,6 +644,50 @@ In the above example, :code:`MultimeterA` and :code:`MultimeterB` use a differen
 :code:`MultimeterB` can be defined subclassing :code:`MultimeterA` and just implementing the difference.
 
 
+.. _channels
+
+Instruments with channels
+=========================
+
+Some instruments, like oscilloscopes and voltage sources, have channels whose commands differ only in the channel name. For this case, we have :class:`~pymeasure.instruments.Channel`, which is similar to :class:`~pymeasure.instruments.Instrument` and its property factories, but does expect an :code:`instrument` instead of an :code:`adapter` as parameter. All the channel communication is routed through the instrument's methods (`write`, `read`, etc.). However, :meth:`Channel.write <pymeasure.instruments.Channel.write>` uses `str.format` to replace any occurence of :code:`"{ch}"` with the channel's name in the command.
+
+.. testcode:: :hide:
+
+    # Behind the scene, load the real Instrument
+    from pymeasure.instruments import Instrument
+    from pymeasure.instruments import Channel
+    from pymeasure.test import expected_protocol
+
+.. testcode::
+
+    class VoltageChannel(Channel):
+        """A channel of the voltage source."""
+
+        voltage = Channel.control(
+			"SOURce{ch}:VOLT?", "SOURce{ch}:VOLT %f",
+			"""The output voltage of this channel, can be set and read.""",
+		)
+
+
+    class ExtremeChannel(Instrument):
```

Comment:
Would InstrumentWithChannels be a better name? It's an instrument, not a channel, so the previous name felt confusing

Reply:
Done